lib/tests/formulae.rb

```diff
@@ -376,7 +376,17 @@ def setup_formulae_deps_instances(formula, formula_name, args:)
 
         @testable_dependents = @source_dependents.select(&:test_defined?)
         @bottled_dependents = with_env(HOMEBREW_SKIP_OR_LATER_BOTTLES: "1") do
-          dependents.select(&:bottled?)
+          if OS.linux? &&
+             tap.present? &&
+             tap.full_name == "Homebrew/homebrew-core" &&
+             ENV["HOMEBREW_SKIP_UNBOTTLED_LINUX_TESTS"]
+            # :all bottles are considered as Linux bottles, but as we did not bottle
+            # everything (yet) in homebrew-core, we do not want to test formulae
+            # with :all bottles for the time being.
```

Comment:
Do we want to add a `TODO` to the comment?

Reply:
We will remove HOMEBREW_SKIP_UNBOTTLED_LINUX_TESTS eventually so this block of code will be cleaned up.